fluent/fluent.go

```diff
@@ -108,6 +113,10 @@ type Fluent struct {
 	closed         bool
 	wg             sync.WaitGroup
 
+	// time (unix milliseconds) at which the most recent connection to fluentd
+	// address was established.
+	latestReconnectTime int64
```

Comment:
This should probably be `time.Time` instead of a Unix millisecond count, since that provides a monotonic clock for the later comparison operations. The check would be something like:

```go
if time.Since(f.latestReconnectTime) > f.AsyncReconnectInterval * time.Millisecond {
    // reconnect
}
```

Reply:
Applied 👍 (just had to transform the int64 AsyncReconnectInterval to a time.Duration). It makes the code cleaner (no need for comments) and uses time native methods, thanks for the suggestion.

On the first tick of the `for` loop in `run` now (if AsyncReconnectInterval is set), the first connection will now be made by the code in the loop, whereas it previously did so in `writeWithRetry`. This is because of the time.Time ["zero" value](https://stackoverflow.com/questions/23051973/what-is-the-zero-value-for-time-time-in-go). We could avoid this by doing `!f.latestReconnectTime.IsZero()` but I don't see the need for this check, as it's the same connection call made, just in a different place, and the condition is only relevant the first run, and would somewhat bloat the code, in my opinion. Just wanted to call it out in case the maintainers here prefer that I do that.

Reply:
Oh, was it failing to compile without the explicit `time.Duration(…)` conversion? [I tried something like that in the playground](https://go.dev/play/p/BPhaK932YKv) and it seemed to work.

Reply:
I think that's because the `reconnectInterval` there is a `const` (an [unsigned integer constant](https://yourbasic.org/golang/gotcha-multiply-duration-integer/)). Where it's an attribute of a struct, like here, it needs to be cast: https://go.dev/play/p/WQP-uCAAh1h

Reply:
Thanks, you're absolutely right, I should have recognized that the difference was the untyped constant.